Source/Core/Common/HookableEvent.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/Logging/Log.h"
+#include "Common/StringLiteral.h"
+
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace Common
+{
+struct HookBase
+{
+  virtual ~HookBase() = default;
+
+protected:
+  HookBase() = default;
+
+  // This shouldn't be copied. And since we always wrap it in unique_ptr, no need to move it either
```

Comment:
Are you relying on the fact that specifying any of the copy-constructor/destructor removes the implicit move operations?  It might be best to specify it directly to sync with the comment:

```
HookBase(HookBase&&) = delete;
HookBase& operator=(HookBase&&) = delete;
```

Reply:
I was; But it's good to be explicit.